cache/Cache.js

```diff
@@ -0,0 +1,63 @@
+const EventEmitter = require('events')
+const _createRedisClient = require('./util/createRedisClient')
+const msgpack = require('msgpack-lite')
+
+// default entry expiration time in seconds
+const DEFAULT_EXPIRATION = 86400
+
+class Cache extends EventEmitter {
+  constructor ({ logger, nodes, defaultTtl, redisClientOptions }) {
+    super()
+    this._logger = logger
+    this._defaultTtl = defaultTtl || DEFAULT_EXPIRATION
+
+    this._onReady = () => this.emit('ready')
+    this._onError = (err) => this.emit('error', err)
+    this._onReconnecting = () => this.emit('reconnecting')
+    this._onNodeError = (err) => this.emit('node error', err)
+
+    const redisClient = this._redisClient =
+      _createRedisClient(nodes, redisClientOptions)
+
+    // forward events to clients so that they
+    // can handle situations appropriately
+    redisClient
+      .on('ready', this._onReady)
+      .on('reconnecting', this._onReconnecting)
+      .on('node error', this._onNodeError)
+      .on('error', this._onError)
+  }
+
+  async get (key) {
+    const encoded = await this._redisClient.getBuffer(key)
```

Comment:
What do you think about adding a `try/catch` around these `await this._redisClient.*` functions and log an error + throw it afterward? Then we wouldn't have to add log errors in all of the services. The logger that is passed could eventually have appenders for different log services (Sumo etc.) or write to file automatically.

```js
async get (key) {
  let encoded
  try {
    encoded = this._redisClient.getBuffer(key)
  } catch (err) {
    // logger.error could have an error appender for Sumo, write to file, etc. We wouldn't have to log out the error
    // everywhere that we use cache service calls
    this._logger.error(`Error occurred fetching key "${key}"`, err)
    throw err
  }
  return encoded ? msgpack.decode(encoded) : encoded
}
```